offscreen_effect_player/offscreen_effect_player.cpp

```diff
@@ -141,65 +100,39 @@ namespace bnb
     {
         auto task = [this, effect = effect_path]() {
             m_ort->activate_context();
-            oep_api::load_effect(effect);
+            m_ep->load_effect(effect);
+            m_ort->deactivate_context();
         };
         m_scheduler.enqueue(task);
     }
 
-    /* offscreen_effect_player::call_js_method */
-    void offscreen_effect_player::call_js_method(const std::string& method, const std::string& param)
+    /* offscreen_effect_player::unload_effect */
+    void offscreen_effect_player::unload_effect()
     {
-        auto task = [this, method = method, param = param]() {
-            m_ort->activate_context();
-            oep_api::call_js_method(method, param);
-        };
-        m_scheduler.enqueue(task);
+        load_effect("");
     }
 
-    /* offscreen_effect_player::read_current_buffer */
-    void offscreen_effect_player::read_current_buffer(std::function<void(bnb::data_t data)> callback)
+    /* offscreen_effect_player::pause */
+    void offscreen_effect_player::pause()
     {
-        if (std::this_thread::get_id() == render_thread_id) {
-            callback(m_ort->read_current_buffer());
-            return;
-        }
-
-        oep_wptr this_ = shared_from_this();
-        auto task = [this_, callback]() {
-            if (auto this_sp = this_.lock()) {
-                callback(this_sp->m_ort->read_current_buffer());
-            }
-        };
-        m_scheduler.enqueue(task);
+        m_ep->pause();
     }
 
-    /* offscreen_effect_player::read_current_buffer */
-    bnb::data_t offscreen_effect_player::read_current_buffer()
+    /* offscreen_effect_player::resume */
+    void offscreen_effect_player::resume()
     {
-        if (std::this_thread::get_id() == render_thread_id) {
-            return m_ort->read_current_buffer();
-        }
-        auto task = [this]() {
-            return this->m_ort->read_current_buffer();
-        };
-        return m_scheduler.enqueue(task).get();
+        m_ep->resume();
     }
 
-    /* offscreen_effect_player::get_current_buffer_texture */
-    void offscreen_effect_player::get_current_buffer_texture(oep_texture_cb callback)
+    /* offscreen_effect_player::call_js_method */
+    void offscreen_effect_player::call_js_method(const std::string& method, const std::string& param)
     {
-        if (std::this_thread::get_id() == render_thread_id) {
-            callback(m_ort->get_current_buffer_texture());
-            return;
-        }
-
-        oep_wptr this_ = shared_from_this();
-        auto task = [this_, callback]() {
-            if (auto this_sp = this_.lock()) {
-                callback(this_sp->m_ort->get_current_buffer_texture());
-            }
+        auto task = [this, method = method, param = param]() {
```

Comment:
it looks like this capture will copy parameters, please review

Reply:
Yes. Parameters passed to function by reference. Lambda is executed async. At the moment of execution, the parameters will be unavailable.